app/slashcommands/command_share.go

```diff
@@ -240,6 +241,15 @@ func (sp *ShareProvider) doInviteRemote(a *app.App, args *model.CommandArgs, mar
 		return responsef("Must specify a valid remote cluster id to invite.")
 	}
 
+	// Check if channel is shared or not.
+	if _, err := a.GetSharedChannel(args.ChannelId); err != nil {
```

Comment:
Will be good to do an existence check here as well, since we explicitly ignoring the result of the query. But logic wise this LGTM

Reply:
I think that should be a test case of the store function itself? Checking for `ErrNotFound` should automatically negate the existence of the channel.

Reply:
I was referring to the underlying query. Since the existing query is doing a `select * from SharedChannels where id = x`. Similar to what we were discussing in a previous PR

Reply:
Oh my bad :facepalm: gotcha .. will look into it.

Reply:
I'm 0/5 on whether the optimized existence check is needed, however I will say that inviting a remote cluster to a channel is a very rare event.